pkg/config.go

```diff
@@ -11,7 +11,39 @@ import (
 
 // SystemConfig is configuration loaded from kubernetes config and secrets that includes information about the
 // database, server, etc.
-type SystemConfig = map[string]string
+type SystemConfig map[string]string
+
+// GetValue returns the value in the underlying map if it exists, otherwise nil is returned
+// If the value does not exist, it is also logged.
+func (s SystemConfig) GetValue(name string) *string {
+	value, ok := s[name]
+	if !ok {
+		log.WithFields(log.Fields{
+			"Method": "SystemConfig.GetValue",
+			"Name":   name,
+			"Error":  "does not exist",
+		})
+
+		return nil
+	}
+
+	return &value
+}
+
+// OnepanelDomain gets the ONEPANEL_DOMAIN value, or nil.
+func (s SystemConfig) OnepanelDomain() *string {
```

Comment:
Let's just call this `Domain`. I think we should also have `FQDN` here too.

Reply:
@rushtehrani Makes sense, and sure thing. I'm slowly adding them as I encounter them in the code.